engine/src/main/java/io/zeebe/engine/processor/workflow/ExpressionProcessor.java

```diff
@@ -58,6 +59,31 @@ public ExpressionProcessor(
         .map(this::wrapResult);
   }
 
+  /**
+   * Evaluates the given expression and returns the result as string wrapped in {@link
+   * DirectBuffer}. If the evaluation fails or the result is not a string then an exception is
+   * thrown.
+   *
+   * @param expression the expression to evaluate
+   * @param scopeKey the scope to load the variables from
+   * @return the evaluation result as buffer
+   * @throws EvaluationException if expression evaluation failed
+   */
+  public DirectBuffer evaluateStringExpression(final Expression expression, final long scopeKey) {
```

Comment:
The caller expects a string instead of a buffer.

```suggestion
  public String evaluateStringExpression(final Expression expression, final long scopeKey) {
```